tools/test.py

```diff
@@ -1699,23 +1719,23 @@ def Main():
       os.makedirs(tempdir)
     except OSError as exception:
       if exception.errno != errno.EEXIST:
-        print "Could not create the temporary directory", options.temp_dir
-        sys.exit(1)
+        print("Could not create the temporary directory " + options.temp_dir)
+        return 1
 
   if options.report:
     PrintReport(all_cases)
 
   result = None
-  def DoSkip(case):
+  def DoKeep(case):
     # A list of tests that should be skipped can be provided. This is
     # useful for tests that fail in some environments, e.g., under coverage.
     if options.skip_tests != [""]:
         if [ st for st in options.skip_tests if st in case.case.file ]:
-            return True
+            return False
     if SKIP in case.outcomes or SLOW in case.outcomes:
-      return True
-    return FLAKY in case.outcomes and options.flaky_tests == SKIP
-  cases_to_run = [ c for c in all_cases if not DoSkip(c) ]
+      return False
+    return not (FLAKY in case.outcomes and options.flaky_tests == SKIP)
+  cases_to_run = filter(DoKeep, all_cases)
```

Comment:
Since you're trying to make the script compatible with both Python 2 and Python 3, I don't think this change makes sense. A list comprehension would work identically in either version, but `filter` is an iterator in Python 3, so it would raise an exception a few lines below at https://github.com/nodejs/node/pull/23251/files#diff-37479d55e637ac7befeefff7f6d88025R1748.